controllers/apply/form-router-next/index.js

```diff
@@ -458,39 +473,51 @@ function initFormRouter({
      * Stream the file's headers and serve it directly as a response
      * @see https://stackoverflow.com/a/43356401
      */
-    router.route('/download/:fieldName/:filename').get((req, res, next) => {
-        const { currentlyEditingId, currentApplicationData } = res.locals;
+    router
+        .route('/download/:fieldName/:filename')
+        .get(async (req, res, next) => {
+            const { currentlyEditingId, currentApplicationData } = res.locals;
 
-        const fileData = currentApplicationData[req.params.fieldName];
-        const matchesField =
-            fileData && fileData.filename === req.params.filename;
+            const fileData = currentApplicationData[req.params.fieldName];
+            const matchesField =
+                fileData && fileData.filename === req.params.filename;
 
-        if (matchesField) {
-            getObject({
-                formId: formId,
-                applicationId: currentlyEditingId,
-                filename: req.params.filename
-            })
-                .on('httpHeaders', (code, headers) => {
-                    res.status(code);
-                    if (code < 300) {
-                        res.set(
-                            pick(
-                                headers,
-                                'content-type',
-                                'content-length',
-                                'last-modified'
-                            )
-                        );
-                    }
-                })
-                .createReadStream()
-                .on('error', next)
-                .pipe(res);
-        } else {
-            next();
-        }
-    });
+            if (matchesField) {
+                const pathConfig = {
+                    formId: formId,
+                    applicationId: currentlyEditingId,
+                    filename: req.params.filename
+                };
+
+                try {
+                    await checkAntiVirus(pathConfig);
+                } catch (err) {
+                    return res.send(
+                        'This file has been blocked for security reasons.'
+                    );
+                }
+
+                getObject(pathConfig)
```

Comment:
I'd possibly move this block inside the `try`

Reply:
👍